src/groovy/au/org/emii/gogoduck/worker/Worker.groovy

```diff
@@ -12,21 +15,30 @@ class Worker {
     Integer fileLimit
     Integer maxGogoduckTimeMinutes
 
+    Integer TOO_MANY_FILES_EXIT_CODE = 3
+
     void run(successHandler, failureHandler) {
+        def startTime = DateTime.now()
+        def process
+
         try {
-            def process = execute(getCmd())
+            process = execute(getCmd())
             log.info("worker output: ${IOUtils.toString(process.getInputStream(), 'UTF-8')}")
 
             if (process.exitValue() == 0) {
                 successHandler(job)
-            }
-            else {
-                failureHandler(job)
+                return
             }
         }
         catch (IOException e) {
```

Comment:
The problem is I will get an IOException when it times out. That's for sure.

Perhaps you're right. I could move the `getReason` code to be also here, then return. Wouldn't bother me.